tests/rbd_mirror/test_rbd_mirror_secondary_full.py

```diff
@@ -0,0 +1,122 @@
+"""Test case covered -
+    CEPH-9507 - Create a mirror image, when secondary cluster
+    doesn't have enough space left to mirror the image copy
+
+    Pre-requisites :
+    1. Two Clusters must be up and running to create pool
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files on each node.
+
+    Test Case Flow:
+        1. Create a pool on both clusters.
+        2. Create an Image on primary mirror cluster in same pool.
+        3. Configure mirroring (peer bootstrap) between two clusters.
+        4. Enable image mode snapshot based mirroring on the pool respectively.
+        5. Start running IOs on the primary image.
+        6. keep on running IOs till secondary cluster becomes full
+        7. verify after secondary becomes full, mirroring should fail
+"""
+
+
+import json
+
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def get_total_avail(cluster):
+    """Method to retrieve total available space in cluster"""
+    out = cluster.exec_cmd(cmd="ceph df -f json", output=True)
+    data = json.loads(out)
+    total_avail_bytes = data["stats"]["total_avail_bytes"]
+    total_avail_gb = total_avail_bytes / (1024**3)
+    return int(total_avail_gb)
+
+
+def run(**kw):
+    """Verification of mirroring failure,
+    when secondary cluster storage is full.
+
+    Args:
+        **kw: test data
+
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+
+    try:
+        log.info(
+            "Starting CEPH-9507 - Verification of mirroring failure "
+            "when secondary cluster storage is full"
+        )
+        rbd = Rbd(**kw)
+        config = kw.get("config")
+        mode = config.get("mode")
+        mirrormode = config.get("mirrormode")
+        poolname = rbd.random_string() + "_test_pool"
+        imagename = rbd.random_string() + "_test_image"
+        imagespec = poolname + "/" + imagename
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd1"), config
+        )
+        mirror2 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd2"), config
+        )
+
+        cluster1_size = get_total_avail(mirror1)
+        cluster2_size = get_total_avail(mirror2)
+
+        if cluster2_size > cluster1_size:
+            log.error("secondary cluster more than primary hence test cannot perform")
```

Comment:
This need not be true. This will hold good only in the case when you are using primary mirroring alone to fill the secondary cluster. But secondary cluster could be filled by any other application running on it, or even by logs or any other reason. 

So this should not be a reason for you to not execute this test case.

Reply:
added this validation thinking only `osd` storage space from primary and secondary cluster.